dev-docs/services/ci/README.md

```diff
@@ -0,0 +1,114 @@
+# The CI Service in Hail Batch
+
+Hail Batch includes a CI service which has three jobs:
+
+- Runs tests against pull requests
+- Merges PRs to the `main` branch
+- Deploys services to the live infrastructure
+
+It does all of these things only if certain conditions are met.
+
+## Deployment
+
+The CI service itself is deployed as a Kubernetes service in the Hail Batch cluster. See 
+[architecture diagram](../Hail%20Batch%20Architectural%20Diagram.png).
+
+As part of its configuration, CI must be configured with an access token allowing it to operate on 
+behalf of a github account called hail-ci-robot.
+
+To make CI work it needs to receive update triggers from the github repository. These are
+configured within the github repository itself.
+
+## Running tests against pull requests
+
+The CI service is constantly monitoring for new PRs in the `hail` repository. 
+
+When a new PR is opened, or an existing PR changes, hooks are triggered which alert the CI service. 
+The CI service will then:
+
+- Check for validity
+  - The author must be a known developer (or else a commit SHA can be individually approved via the CI service UI)
+  - The PR must be against the `main` branch
+- Run tests
+  - The CI service will run tests against the PR
```

Comment:
Might be useful to document that it squashes and rebases the PR on top of `main` and then runs tests. I've experienced strange linting/formatting failures because I've generally used merge commits rather than rebasing.

Reply:
👍 I'll add this under the details of what the test batch does below.